packages/local_auth/local_auth_windows/windows/local_auth_plugin.cpp

```diff
@@ -0,0 +1,223 @@
+// found in the LICENSE file.
+#include "local_auth.h"
+
+namespace local_auth_windows {
+
+using namespace flutter;
+using namespace winrt;
+
+template <typename T>
+// Helper method for getting an argument from an EncodableValue
+T GetArgument(const std::string arg, const EncodableValue* args, T fallback) {
+  T result{fallback};
+  const auto* arguments = std::get_if<EncodableMap>(args);
+  if (arguments) {
+    auto result_it = arguments->find(EncodableValue(arg));
+    if (result_it != arguments->end()) {
+      result = std::get<T>(result_it->second);
+    }
+  }
+  return result;
+}
+
+// Returns the window's HWND for a given FlutterView
+HWND GetRootWindow(flutter::FlutterView* view) {
+  return ::GetAncestor(view->GetNativeWindow(), GA_ROOT);
+}
+
+// Converts the given UTF-8 string to UTF-16.
+std::wstring Utf16FromUtf8(const std::string& utf8_string) {
+  if (utf8_string.empty()) {
+    return std::wstring();
+  }
+  int target_length =
+      ::MultiByteToWideChar(CP_UTF8, MB_ERR_INVALID_CHARS, utf8_string.data(),
+                            static_cast<int>(utf8_string.length()), nullptr, 0);
+  if (target_length == 0) {
+    return std::wstring();
+  }
+  std::wstring utf16_string;
+  utf16_string.resize(target_length);
+  int converted_length =
+      ::MultiByteToWideChar(CP_UTF8, MB_ERR_INVALID_CHARS, utf8_string.data(),
+                            static_cast<int>(utf8_string.length()),
+                            utf16_string.data(), target_length);
+  if (converted_length == 0) {
+    return std::wstring();
+  }
+  return utf16_string;
+}
+
+class UserConsentVerifierImpl : public UserConsentVerifier {
+ public:
+  explicit UserConsentVerifierImpl(std::function<HWND()> window_provider)
+      : get_root_window_(std::move(window_provider)){};
+  virtual ~UserConsentVerifierImpl() = default;
+
+  Windows::Foundation::IAsyncOperation<
+      Windows::Security::Credentials::UI::UserConsentVerificationResult>
+  RequestVerificationForWindowAsync(std::wstring localizedReason) override {
+    auto userConsentVerifierInterop = winrt::get_activation_factory<
+        Windows::Security::Credentials::UI::UserConsentVerifier,
+        IUserConsentVerifierInterop>();
+
+    auto hWnd = get_root_window_();
+
+    HSTRING hReason;
+    if (WindowsCreateString(localizedReason.c_str(),
+                            static_cast<uint32_t>(localizedReason.size()),
+                            &hReason) != S_OK) {
+      return Windows::Security::Credentials::UI::UserConsentVerificationResult::
+          Canceled;
+    }
+
+    auto consentResult =
+        co_await winrt::capture<Windows::Foundation::IAsyncOperation<
+            Windows::Security::Credentials::UI::UserConsentVerificationResult>>(
+            userConsentVerifierInterop,
+            &::IUserConsentVerifierInterop::RequestVerificationForWindowAsync,
+            hWnd, hReason);
+
+    WindowsDeleteString(hReason);
```

Comment:
Is there a unique_ptr adapter for this out of the box that we could use?

Reply:
Yes, there is, I'll switch to it.